pkg/services/librarypanels/database.go

```diff
@@ -144,7 +141,7 @@ func (lps *LibraryPanelService) disconnectLibraryPanelsForDashboard(dashboardID
 		if rowsAffected, err := result.RowsAffected(); err != nil {
 			return err
 		} else if rowsAffected != panelCount {
-			return fmt.Errorf("disconnected %d panels, while expecting %d", rowsAffected, panelCount)
+			lps.log.Warn("Number of disconnects does not match number of panels", "dashboard", dashboardID, "rowsAffected", rowsAffected, "panelCount", panelCount)
```

Comment:
I think the only condition that should be logged is _if there are connections remaining in the DB_, we shouldn't worry the Grafana admin if we removed more connections than expected. It's not a blocker though, so I'll approve.

Reply:
@aknuds1 the check for the remaining connections would have to be done in a separate transaction, correct?

Reply:
I don't think testing in another transaction would be worth it, not catch all cases.

Reply:
But wouldn't the count always be `0` within the transaction?

Reply:
True, within the transaction there shouldn't be any more connections I think.

Reply:
I think I'll leave at as is for now.

Reply:
It's still subject to a race condition though, since the panel count may have become stale before the transaction starts.